cli/internal/cmd/upgradeapply.go

```diff
@@ -271,9 +271,10 @@ func (u *upgradeApplyCmd) migrateTerraform(cmd *cobra.Command, conf *config.Conf
 			}
 			if !ok {
 				cmd.Println("Aborting upgrade.")
-				// Remove the upgrade directory
-				if err := u.fileHandler.RemoveAll(upgradeDir); err != nil {
-					return res, fmt.Errorf("cleaning up upgrade directory %s: %w", upgradeDir, err)
+				// User doesn't expect to see any changes in his workspace after aborting an "upgrade apply",
+				// therefore, roll back to the backed up state.
+				if err := u.clusterUpgrader.RollbackClusterWorkspace(); err != nil {
+					u.log.Debugf("Failed to rollback Terraform workspace: %s", err)
```

Comment:
We should return an error if the rollback fails, or at least use `cmd.PrintErr`, so the error is always logged, not only with debug output enabled.
In case restoring the backup fails, we should also tell the user where the backup is located so they may restore it manually.